src/Parser/DecimalMoneyParser.php

```diff
@@ -24,7 +24,7 @@
  */
 final class DecimalMoneyParser implements MoneyParser
 {
-    public const DECIMAL_PATTERN = '/^(?P<sign>-)?(?P<digits>0|[1-9]\d*)?\.?(?P<fraction>\d+)?$/';
+    public const DECIMAL_PATTERN = '/^(?P<sign>-)?(?P<digits>[0-9]\d*)?\.?(?P<fraction>\d+)?$/';
```

Comment:
The regexp can be simplified this way:

```suggestion
    public const DECIMAL_PATTERN = '/^(?P<sign>-)?(?P<digits>\d+)?\.?(?P<fraction>\d+)?$/';
```